### ISSUE TITLE
Annoying warning in 3.1

### ISSUE THREAD

**Issue by netfenceventu** [psy0rz/zfs_autobackup#101]
Hello.

I've been using zfs_autobackup to just take snapshot locally (without transferring to/from anywhere else).
I'm running it from cron hourly as:
zfs-autobackup --keep-source 10,1h1d,1d1w,1w1m,1m1y auto

Starting from version 3.1, I get this warning:
>  NOTE: Source and target are on the same host, excluding received datasets from selection.

(Of course this means getting a mail every hour).

This obviously come from lines 487-493 of ZfsAutobackup.py:

>  if self.args.ssh_source == self.args.ssh_target:
> ... 
>                    self.warning("Source and target are on the same host, excluding received datasets from selection.")

Since I'm not using ssh, both arguments are None: perhaps this should be tested before emitting the warning?
I'm not sure whether exclude_received should be set to True anyway, though.

Thanks.

**Comment by psy0rz** [comment]
The message is valid, but perhaps it should only be a NOTE: in verbose mode, instead of a warning.

If you run zfs-autobackup in snapshot mode on the same source and target host, it has to guess which datasets (if any) are the existing backups.

Since it doesnt know the target_path, it will just assume that all datasets that have a received-property are part of the backup. (since its possible there is another cronjob that makes the actual backup)

tldr: If we wouldn't do this, its possible to create backups of previous backups, forever.

However its not a perfect system, i think we need something better way to recognize  datasets that are already a backup and should be excluded. (in a same-host situation)

**Comment by psy0rz** [comment]
The situation is a bit different then what i explained in the prevous comment:

When the source and target are on the same host, then:

 * In case of snapshot only mode we have to use --exclude-recieved, since we dont know what the target is and we dont want to accidently snapshot it.  We should explain this better in the documentation: #105 (e.g. DO specify the target and use --no-send might be better sometimes)
 * In case of normal backup mode, we use --exclude-path for the target_path, which is fine. (You never want to backup something from that target to that same target)

**Comment by olemmer** [comment]
@netfenceventu (and @all coming from google search :-) )

I completely understand your problem of getting a mail every hour when zfs-autobackup is set up as snapshot tool in cron.
Even with `--exclude-received` I still get the annoying NOTE in non-verbose mode.

This solved the problem for hourly snapshots in cron:
`--ssh-target <dummy name> --no-send`

**Comment by psy0rz** [comment]
Thanks for the workaround. Its already fixed in 3.2, but thats still in alpha stadium.

Also it shouldnt give a warning or even a note when --exclude-received, so thats another thing i will fix rightaway.
